packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -25,40 +30,48 @@ export class RootDataObjectWithChildDataObject extends DataObjectWithCounter {
 
     protected async hasInitialized(): Promise<void> {
         await super.hasInitialized();
+        this.uniqueId = makeRandom().uuid4();
         await this.createAndReferenceChild();
-        // TODO: Start logic goes here after we agree on what it should be.
+        this.start();
     }
 
-    // TODO: Make private - left public so the build passes
-    public async unreferenceChild() {
+    private unreferenceChild() {
         assert(this.child !== undefined, `Child should be defined when unreferencing!`);
-        this.unreferencedChild = this.child;
         this.child.stop();
         this.root.delete(this.childKey);
-        this.unrefTimeStampMs = Date.now();
     }
 
     private async createAndReferenceChild() {
         this.child = await dataObjectWithCounterFactory.createInstance(this.context.containerRuntime);
-        await this.referenceChild(this.child.handle);
+        this.root.set(this.childKey, this.child.handle);
+        this.child.start();
     }
 
-    private async referenceChild(handle: IFluidHandle<DataObjectWithCounter>) {
-        this.unreferencedChild = this.child;
-        this.root.set(this.childKey, handle);
-    }
+    protected async run() {
+        assert(this.isRunning === true, "Should be running to send ops");
+        // The ideal loop is to send a number of counter ops, and then reference or unreference a child.
+        while (this.isRunning && !this.disposed) {
+            let opsPerformed = 0;
+            // This inner loop generates a number of ops set by this.opsToWait.
+            while (opsPerformed < this.opsToWait) {
+                this.counter.increment(1);
+                // This data is local and allows us to understand the number of changes a local client has created
+                opsPerformed++;
+                await delay(this.delayPerOpMs);
 
-    // This should only be called when inactiveObjectX time isn't greater than unrefTimeStampMs
-    // TODO: Make private - left public so the build passes
-    public async reviveChild() {
-        assert(this.unreferencedChild !== undefined, `An old handle should exist when reviving`);
-        assert(this.unrefTimeStampMs !== undefined, `An unreferenceTimestamp should exist when reviving`);
-        await this.referenceChild(this.unreferencedChild.handle);
-        this.unreferencedChild = undefined;
-        this.unrefTimeStampMs = undefined;
-    }
+                if (!this.isRunning || this.disposed) {
```

Comment:
Totally optional - I think you can avoid duplicating the break condition by defining `opsPerformed` outside the main while loop, and then rather than an inner loop. only do the reference/unreference step if  `opsPerformed % this.opsToWait === 0` or something.